### ISSUE TITLE
argocd-dex not using logformat/loglevel from cli

### ISSUE THREAD

**Issue by hermanbanken** [argoproj/argo-cd#19884]
## What is the problem
ArgoCD (argocd-dex) starts `dex` with a YAML configuration from [argocd_dex.go#L100](https://github.com/argoproj/argo-cd/blob/bc4c4757fddbb64a0e119372cad16cc29248ed97/cmd/argocd-dex/commands/argocd_dex.go#L100). Before starting dex, it parses cli arguments `logformat` and `loglevel` :
https://github.com/argoproj/argo-cd/blob/bc4c4757fddbb64a0e119372cad16cc29248ed97/cmd/argocd-dex/commands/argocd_dex.go#L139-L140
into cmd/util
https://github.com/argoproj/argo-cd/blob/bc4c4757fddbb64a0e119372cad16cc29248ed97/cmd/util/common.go#L3-L6

However, the dex YAML configuration parses environment variables defined in `common` anew (`ARGOCD_LOG_LEVEL`, `ARGOCD_LOG_FORMAT`):
https://github.com/argoproj/argo-cd/blob/bc4c4757fddbb64a0e119372cad16cc29248ed97/util/dex/config.go#L52-L55

These sets of arguments/variables are misaligned.

## What we observe

Our deployment YAML contains:
```yaml
      - args:
        - rundex
        command:
        - /shared/argocd-dex
        - --logformat=json
        - --loglevel=info
```

We expected this would log everything as json, and at info level. However, the logs contain the following:

```text
{"level":"info","msg":"dex config unmodified","time":"2024-09-10T09:22:42Z"}                                                                                                                        
{"level":"info","msg":"Notifying 1 settings subscribers: [0xc00123cea0]","time":"2024-09-10T09:22:47Z"}                                                                                                                                                                                                           
time="2024-09-10T12:30:15Z" level=info msg="keys expired, rotating"                                                                                                                                 
time="2024-09-10T12:30:15Z" level=info msg="keys rotated, next rotation: 2024-09-10 18:30:15.78678407 +0000 UTC"                                                                                    
time="2024-09-10T18:30:17Z" level=info msg="keys expired, rotating"                                                                                                                                 
time="2024-09-11T06:30:19Z" level=info msg="keys rotated, next rotation: 2024-09-11 12:30:19.81644692 +0000 UTC"                                                                                    
{"level":"info","msg":"Notifying 1 settings subscribers: [0xc00123cea0]","time":"2024-09-11T08:00:23Z"}                                                                                             
{"level":"info","msg":"dex config unmodified","time":"2024-09-11T08:00:23Z"}                                                                                                                        
{"level":"info","msg":"Notifying 1 settings subscribers: [0xc00123cea0]","time":"2024-09-11T08:00:27Z"}                                                                                             
{"level":"info","msg":"dex config unmodified","time":"2024-09-11T08:00:27Z"}                                                                                                                        
time="2024-09-11T08:00:43Z" level=info msg="login successful: connector \"github\", username=\"Herman\", preferred_username=\"hermanbanken\", email=\"redacted\", groups=[\"redacted
time="2024-09-11T08:00:47Z" level=info msg="login successful: connector \"github\", username=\"Herman\", preferred_username=\"hermanbanken\", email=\"redacted\", groups=[\"redacted
{"level":"info","msg":"Notifying 1 settings subscribers: [0xc00123cea0]","time":"2024-09-11T08:12:15Z"}                                                                                             
```

This contains non-json logs on the stderr stream, resulting in non-errors reported as errors.

**Comment by hermanbanken** [comment]
Sorry! Duplicate of #13191. We hadn't updated yet.
